fission/spec.go

```diff
@@ -960,31 +960,11 @@ func applyResources(fclient *client.Client, specDir string, fr *FissionResources
 // localArchiveFromSpec creates an archive on the local filesystem from the given spec,
 // and returns its path and checksum.
 func localArchiveFromSpec(specDir string, aus *ArchiveUploadSpec) (*fission.Archive, error) {
-	// get root dir
-	var rootDir string
-	if len(aus.RootDir) == 0 {
-		rootDir = filepath.Clean(specDir + "/..")
-	} else {
-		rootDir = aus.RootDir
-	}
-
-	// get a list of files from the include/exclude globs.
-	//
-	// XXX if there are lots of globs it's probably more efficient
-	// to do a filepath.Walk and call path.Match on each path...
 	files := make([]string, 0)
-	if len(aus.IncludeGlobs) == 1 && archiver.Zip.Match(aus.IncludeGlobs[0]) {
-		files = append(files, aus.IncludeGlobs[0])
-	}
-	for _, relativeGlob := range aus.IncludeGlobs {
```

Comment:
The root cause of the issue is here. We can solve the issue by adding an `else` here.

Reply:
Thanks - good catch - I have fixed and updated the PR @life1347